### PR TITLE
:memo: Improves README.md

### PR DESCRIPTION
**Pull request by PedroAugustoRamalhoDuarte**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,92 +29,100 @@ Or install it yourself as:
...
+```
 
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can
+also run `bin/console` for an interactive prompt that will allow you to experiment.
 
-To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then run `bundle exec rake release`, which will create a git tag for the version, push git commits and the created tag, and push the `.gem` file to [rubygems.org](https://rubygems.org).
+To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the
```
Comment (JoaoPedroAssis) [inline README.md:124]: To install this gem in your local machine, run `bundle exec rake install` and update the version number in `version.rb`, if you wish to release a new version. After that, run `bundle exec rake release`, which will create a git tag, push it alongside your commits and publish the `.gem` file into [rubygems.org](https://rubygems.org).

File: README.md [hunk 2]
```diff
@@ -30,92 +29,100 @@ Or install it yourself as:
...
-Bug reports and pull requests are welcome on GitHub at https://github.com/[USERNAME]/rest-api-generator. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [code of conduct](https://github.com/[USERNAME]/rest-api-generator/blob/master/CODE_OF_CONDUCT.md).
+Bug reports and pull requests are welcome on GitHub at https://github.com/SwitchDreams/rest-api-generator. This project
+is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to
+the [code of conduct](https://github.com/SwitchDreams/rest-api-generator/blob/master/CODE_OF_CONDUCT.md).
 
 ## License
 
 The gem is available as open source under the terms of the [MIT License](https://opensource.org/licenses/MIT).
 
 ## Code of Conduct
```
Comment (JoaoPedroAssis) [inline README.md:138]: Acho que pode tirar essa parte a não ser que o code of conduct seja melhor explicado, pq já tem la em cima a mesma coisa

**Review by JoaoPedroAssis — CHANGES_REQUESTED** [review]
Brabíssimo, só avalia essas changes de ingles e tentei deixar mais parecido com a linguagem que eu vejo nos repos por aí

File: README.md [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
-    <li> Factory bot </li>
-</ul>
+## How it works?
+
+The gems use rails generators and some templates to create all resources needed to build an REST API.
 
-the commnad its simmilar to the model generator "rails g model model_name attributes", in fact it invokes this generator to genrate the model and migration file
+The controller is built following SwitchDreams default way:
+
+- Using an error handler module to deal with a custom exception for centralize error handler following
```
[the remark this answers is not in the document]
Comment (PedroAugustoRamalhoDuarte) [reply README.md:11]: This one does not shows this gem feelings we do not capture all errors

[the remark this answers is not in the document]
Comment (PedroAugustoRamalhoDuarte) [reply README.md:11]: Maybe this is better?
We use an error module to centralize error handler, rescuing from a custom and some active record exceptions.
  The inspiration for this strategy was
  this [article](https://medium.com/rails-ember-beyond/error-handling-in-rails-the-modular-way-9afcddd2fe1b.)

File: README.md [hunk 4]
```diff
@@ -1,17 +1,16 @@
...
+  this article https://medium.com/rails-ember-beyond/error-handling-in-rails-the-modular-way-9afcddd2fe1b.
+- For tests, we use RSpec and FactoryBot.
```
[the remark this answers is not in the document]
Comment (PedroAugustoRamalhoDuarte) [reply README.md:13]: Its RSpec the gem name, i will change in other context

File: README.md (hunk 1 above)
Comment (PedroAugustoRamalhoDuarte) [reply README.md:124]: This section was automatic generated, i will keep the same

**Pull request merged**